Spring-Boot/Spring-Boot-Submission/src/test/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionControllerTest.java

```diff
@@ -257,4 +259,40 @@ private SubmissionView getBestSubmission(int studentId) throws Exception {
                 .andExpect(status().isOk()), SubmissionView.class);
     }
 
+    @Test
+    void GivenAStudentHasOneSubmission_WhenRejudgeThatSubmission_ThenTheStudentShouldHaveTwoSubmissionsWithTheSameFileIds() throws Exception {
+
+        var submission = submitCodeAndGet(ADMIN_ID, ADMIN_TOKEN);
+
+        mockMvc.perform(post("/api/submissions/{submissionId}/judge", submission.getId()))
+                .andExpect(status().isOk());
+
+        allOfTheStudentSubmissionsShouldHaveTheSameFileIds(ADMIN_TOKEN);
+    }
+
+    @Test
+    void GivenTwoSubmissionsOfAQuestionFrom2Admins_WhenRejudgeAllSubmissionsOfThatQuestion_SubmissionsOfBothAdminsShouldHaveTheSameFileIds() throws Exception {
+
+        submitCodeAndGet(ADMIN_ID, ADMIN_TOKEN, codes1);
+        submitCodeAndGet(ADMIN_ID2, ADMIN_TOKEN2, codes2);
+
+        var request = new Request(problem.getId(), new Bag("examId", "1"));
+        mockMvc.perform(post("/api/submissions/judges")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(toJson(request)))
+                .andExpect(status().isOk());
+
+        allOfTheStudentSubmissionsShouldHaveTheSameFileIds(ADMIN_TOKEN);
+        allOfTheStudentSubmissionsShouldHaveTheSameFileIds(ADMIN_TOKEN2);
+    }
```

Comment:
Sorry, the test is still not strict enough.

You should test whether the judger has been deployed